src/test/java/com/timgroup/statsd/TelemetryTest.java

```diff
@@ -0,0 +1,357 @@
+    }
+
+    @Test(timeout = 5000L)
+    public void telemetry_flushInterval() throws Exception {
+        client.telemetry.reset();
+        client.telemetry.incrMetricsSent(1);
+
+        assertThat(client.telemetry.metricsSent.get(), equalTo(1));
+
+        // Start flush timer with a 50ms interval
+        client.telemetry.start(50L);
+
+        // Wait for the flush to happen
+        while (client.telemetry.metricsSent.get() != 0) {
+            try {
+                Thread.sleep(30L);
+            } catch (InterruptedException e) {}
+        }
+        client.telemetry.stop();
+
+        assertThat(client.telemetry.metricsSent.get(), equalTo(0));
+
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.metrics:1|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.events:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.service_checks:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.bytes_sent:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.bytes_dropped:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.packets_sent:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.packets_dropped:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+        assertThat(fakeSender.messages, hasItem("my.prefix.datadog.dogstatsd.client.packets_dropped_queue:0|c|#test,client:java,"
+                    + NonBlockingStatsDClient.JAVA_CLIENT_VERSION_TAG + ",client_transport:udp"));
+    }
+
+    @Test(timeout = 5000L)
+    public void telemetry_droppedData() throws Exception {
+        clientError.telemetry.reset();
+
+        clientError.gauge("gauge", 24);
+
+        // leaving time to the server to flush metrics
+        while (clientError.telemetry.metricsSent.get() == 0
+               || clientError.telemetry.packetsDropped.get() == 0
+               || clientError.telemetry.bytesDropped.get() == 0) {
+            try {
+                Thread.sleep(50L);
```

Comment:
Am I correct to assume that you're doing this because you enabled the telemetry to flush in 50ms intervals here: https://github.com/DataDog/java-dogstatsd-client/pull/97/files#diff-bedc79efc014c08d01856bdf92f57801R288 ? If so, we're making the assumption the tests will run in order, which may be a safe assumption but is something I would prefer we did not assume in any case. If every test can be self-contained, that would be best.

Reply:
No we're actually waiting for the whole process to flush metrics. This is the same than when we wait for messages in our DummyServer https://github.com/DataDog/java-dogstatsd-client/blob/master/src/test/java/com/timgroup/statsd/DummyStatsDServer.java#L61.

Tests are unrelated from one another (one use `client` and the other `clientError` also).